> Now that PPoly has the CuPy delegation set up, I wonder if Pchip, being a subclass, just works with CuPy, too. (And if it does, so do Akima, CubicHermiteSpline and CubicSpline). Could you check it here?

For Pchip, it seems to fail for test_roots(). 

```bash
../../../../../.pixi/envs/cupy/lib/python3.13/site-packages/cupyx/scipy/interpolate/_interpolate.py:1305: NotImplementedError
================================================================================= short test summary info ==================================================================================
FAILED scipy/interpolate/tests/test_polyint.py::TestPCHIP::test_roots[cupy] - NotImplementedError: At the moment there is not a GPU implementation for solve
=============================================================================== 1 failed, 8 passed in 2.45s ================================================================================
```

I tried quickly for the others, and I am getting
```bash
FAILED scipy/interpolate/tests/test_polyint.py::TestCubicSpline::test_periodic_eval[cupy] - TypeError: Implicit conversion to a NumPy array is not allowed. Please use `.get()` to construct a NumPy array explicitly.
```
Should we do the others here or later?

--- a/scipy/interpolate/_cubic.py
+++ b/scipy/interpolate/_cubic.py
@@ -172,7 +172,7 @@ class CubicHermiteSpline(PPoly):
 
 
 @xp_capabilities(
-    cpu_only=True, jax_jit=False,
+    cpu_only=True, jax_jit=False, exceptions=["cupy"],
     skip_backends=[
         ("dask.array",
          "https://github.com/data-apis/array-api-extra/issues/488")

--- a/scipy/interpolate/tests/test_polyint.py
+++ b/scipy/interpolate/tests/test_polyint.py
@@ -665,6 +665,7 @@ class TestPCHIP:
             xp.asarray([[3.5], [1.]], dtype=xp.float64)
         )
 
+    @skip_xp_backends("cupy", reason="no CuPy implementation of 'solve'.")
     def test_roots(self, xp):
         # regression test for gh-6357: .roots method should work
         p = pchip(xp.asarray([0, 1]), xp.asarray([-1, 1]))

